AssertionError [ERR_ASSERTION]: Expected values to be strictly equal:
+ actual - expected

+ 'EAI_FAIL'
- 'ENOTFOUND'
    at Socket.<anonymous> (/builds/vbatrla/node_12.x/components/nodejs12/build/a
md64/test/parallel/test-net-dns-error.js:37:10)
    at Socket.<anonymous> (/builds/vbatrla/node_12.x/components/nodejs12/build/a
md64/test/common/index.js:375:15)
    at Socket.emit (events.js:223:5)
    at GetAddrInfoReqWrap.emitLookup [as callback] (net.js:1037:12)
    at GetAddrInfoReqWrap.onlookup [as oncomplete] (dns.js:60:17) {
  generatedMessage: true,
  code: 'ERR_ASSERTION',
  actual: 'EAI_FAIL',
  expected: 'ENOTFOUND',
  operator: 'strictEqual'
}
Command: out/Release/node /builds/vbatrla/node_12.x/components/nodejs12/build/amd64/test/parallel/test-net-dns-error.js

diff -ru node-12.15.0.orig/test/parallel/test-net-dns-error.js node-12.15.0/test/parallel/test-net-dns-error.js
--- node-12.15.0.orig/test/parallel/test-net-dns-error.js	2020-02-06 00:20:35.000000000 +0000
+++ node-12.15.0/test/parallel/test-net-dns-error.js	2020-02-11 21:31:17.680906791 +0000
@@ -26,7 +26,9 @@
 const net = require('net');
 
 const host = '*'.repeat(64);
-const errCode = common.isOpenBSD ? 'EAI_FAIL' : 'ENOTFOUND';
+// Maximum length of a host name is 63 characters. An attempt to resolve longer
+// host name returns permanent failure (EAI_FAIL) on some OS.
+const errCode = common.isOpenBSD || common.isSunOS ? 'EAI_FAIL' : 'ENOTFOUND';
 
 const socket = net.connect(42, host, common.mustNotCall());
 socket.on('error', common.mustCall(function(err) {
